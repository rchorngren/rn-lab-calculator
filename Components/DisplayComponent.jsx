import React from "react";
import { Text, View, StyleSheet } from "react-native";

//using props rather than context
const DisplayComponent = (props) => {
    const styles = StyleSheet.create({
        displayContainer: {
            width: "100%",
            height: 75,
            justifyContent: "center",
            alignItems: "flex-end",
            borderRadius: 5,
            borderWidth: 3,
            borderColor: "black",
            marginBottom: 20,
            overflow: 'hidden',
            backgroundColor: "green",
        },
        displayText: {
            fontSize: 32,
            fontWeight: "bold",
            marginRight: 15,
        },
    });

    return (
        <View style={styles.displayContainer}>
            <Text style={styles.displayText}>{props.displayText}</Text>
        </View>
    )
}

export default DisplayComponent;