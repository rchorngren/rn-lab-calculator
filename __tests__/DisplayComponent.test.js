import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DisplayComponent from '../Components/DisplayComponent';

it("Matches snapshot on DisplayComponent", () => {

    const tree = renderer.create(<DisplayComponent />).toJSON();
    expect(tree).toMatchSnapshot();

});
