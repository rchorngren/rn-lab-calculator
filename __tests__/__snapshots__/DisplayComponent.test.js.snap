// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot on DisplayComponent 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "green",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 3,
      "height": 75,
      "justifyContent": "center",
      "marginBottom": 20,
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
        "fontWeight": "bold",
        "marginRight": 15,
      }
    }
  />
</View>
`;
